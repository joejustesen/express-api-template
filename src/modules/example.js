export default async function exampleGet() {
  const data = {
    bird: 'robin',
    gender: 'female',
    when: new Date().toISOString(),
  };

  return data;
}
