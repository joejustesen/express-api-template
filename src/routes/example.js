import express from 'express';
import asyncHandler from 'express-async-handler';

import handler from '../modules/example';

const router = express.Router();

router.get('/', asyncHandler(handler));
export default router;
