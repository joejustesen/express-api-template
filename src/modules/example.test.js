import example from './example';

describe('/example api conformance tests', () => {
  it('/example should return bird data', async () => {
    const data = await example();
    expect(data).toMatchSnapshot({ when: expect.any(String) });
  });
});
