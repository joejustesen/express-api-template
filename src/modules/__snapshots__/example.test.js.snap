// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/example api conformance tests /example should return bird data 1`] = `
Object {
  "bird": "robin",
  "gender": "female",
  "when": Any<String>,
}
`;
